Solaris linker doesn't like --version-script and maps
with some symbols not being specified.

--- libproxy-0.4.16/libproxy/cmake/libproxy.cmk
+++ libproxy-0.4.16/libproxy/cmake/libproxy.cmk
@@ -21,7 +21,7 @@ set_property(SOURCE ${CMAKE_CURRENT_SOUR
 set_target_properties(libproxy PROPERTIES PREFIX "" VERSION 1.0.0 SOVERSION 1)
 set_target_properties(libproxy PROPERTIES INTERFACE_LINK_LIBRARIES "")
 set_target_properties(libproxy PROPERTIES LINK_INTERFACE_LIBRARIES "")
-if(NOT APPLE)
+if(NOT APPLE AND NOT CMAKE_SYSTEM_NAME MATCHES "SunOS")
   set_target_properties(libproxy PROPERTIES LINK_FLAGS "-Wl,--version-script,${CMAKE_CURRENT_SOURCE_DIR}/libproxy.map")
 endif()
 install(TARGETS libproxy DESTINATION ${LIB_INSTALL_DIR})
